polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -590,8 +588,15 @@ class LoopInfo {
     if (lowerBound)
       return lowerBound;
     OpBuilder builder(loop);
-    lowerBound = getValueOrCreateConstantIndexOp(builder, loop.getLoc(),
-                                                 *loop.getSingleLowerBound());
+    if (loop.getSingleLowerBound().has_value()) {
+      lowerBound = getValueOrCreateConstantIndexOp(builder, loop.getLoc(),
+                                                   *loop.getSingleLowerBound());
+    } else {
+      auto affineLoop = cast<affine::AffineForOp>(loop);
```

Comment:
We interested only in AffineForOp that have a single lower bound (actually  do not think an affine.for can have multiple IVs/LBs/UBs like an affine.parallel). In any case I'd say we are only interested in loops that have a single IV/LB/UB/STEP.

Reply:
The code is confusing because `getSingleLowerBound`:
   - returns the actual LB for `scf:for`
   - return nullopt if the LB is not a constant for `affine::for` (making it seem like the loop has more than one lower bound)

I think the implementation of `getSingleLowerBound` for affine::for is misleading and should be fixed upstream. However we can change this code to make it more understandable to the casual reader. Something like this: 

```
TypeSwitch<LoopLikeOpInterface>(loop).Case<scf::ForOp>(auto scfLoop) {
    // compute lowerBound. 
}).Case<affine::AffineForOp>(auto affineLoop) {
    // document that  loop.getSingleLowerBound()  return nullopt if the LB is not a constant
    .... 
});